packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -432,3 +434,23 @@ export function getAnyType(): SdkBuiltInType {
     encode: "string",
   };
 }
+
+export function filterApiVersionsInEnum(
+  context: TCGCContext, client: SdkClient, sdkVersionsEnum: SdkEnumType
+): void {
+  // if they explicitly set an api version, remove previous versions
+  if (
+    context.apiVersion !== undefined &&
+    context.apiVersion !== "latest" &&
+    context.apiVersion !== "all"
+  ) {
+    const index = sdkVersionsEnum.values.findIndex((v) => v.value === context.apiVersion);
+    if (index >= 0) {
+      sdkVersionsEnum.values = sdkVersionsEnum.values.slice(0, index + 1);
+    }
+  }
+  const defaultApiVersion = getDefaultApiVersion(context, client.service);
```

Comment:
Is the `defaultApiVersion` not the latest item of `sdkVersionsEnum`? We may not need to  retrieve again from service?

Reply:
@chunyu3 we need to use the output from `defaultApiVersion` before we create our `sdkVersionsEnum`, since we use it to create the enum. Would be nice if we could remove it, but I don't think we can